File: code_examples/5-Holding_Data/4-Associative_Array/Java/Main.java
/*
 * This program uses a map
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

import java.util.Scanner;
import java.util.HashMap;
import java.util.Map;

public class Main {
  public static void main(String[] args) {
    // this function uses a uses a map

    // this function uses a map
    String airportName;

    // an empty map
    Map<String, String> airports = new HashMap<>();

    // a map filled with data
    Map<Character, Integer> someInfo = new HashMap<>();
    someInfo.put('a', 1);
    someInfo.put('b', 2);
    someInfo.put('c', 3);

    // adding items
    airports.put("YYZ", "Toronto Pearson");
    airports.put("YOW", "Ottawa Canada");
    airports.put("DUB", "Dublin Ireland");
    airports.put("LHR", "London Heathrow");

    System.out.println("All the airports:");
    for (Map.Entry<String, String> entry : airports.entrySet()) {
        System.out.println("The airport code is " + entry.getKey() + " for " + entry.getValue());
    }
    System.out.println();

    Scanner scanner = new Scanner(System.in);
    System.out.print("Type in an airport code: ");
    airportName = scanner.nextLine();
    System.out.println();

    if (airports.containsKey(airportName)) {
        System.out.println("The name of the airport you chose is " + airports.get(airportName));
    } else {
        System.out.println("That airport is not in the airport's dictionary.");
    }

    // close the Scanner object
    scanner.close();
    System.out.println("\nDone.");
  }
}

File: code_examples/5-Holding_Data/4-Associative_Array/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a map
 */

package main

import (
	"fmt"
)

func main() {
	// this function uses a map

	var airportName string

	// an empty map
	airports := make(map[string]string)

	/*
	// a map filled with data
	someInfo := map[byte]int{
		'a': 1,
		'b': 2,
		'c': 3,
	}
	*/

	// adding items
	airports["YYZ"] = "Toronto Pearson"
	airports["YOW"] = "Ottawa Canada"
	airports["DUB"] = "Dublin Ireland"
	airports["LHR"] = "London Heathrow"

	fmt.Println("All the airports:")
	for code, name := range airports {
		fmt.Printf("The airport code is %s for %s\n", code, name)
	}
	fmt.Println()

	fmt.Print("Type in an airport code: ")
	fmt.Scanln(&airportName)
	fmt.Println()

	if name, ok := airports[airportName]; ok {
		fmt.Println("The name of the airport you chose is", name)
	} else {
		fmt.Println("That airport is not in the airport's dictionary.")
	}

	fmt.Println("\nDone.")
}
